### PR TITLE
Added `ArrayString::from` constructor that takes string slice.

### PR DESCRIPTION
**Pull request by alexreg**
Also added corresponding test.

### CODE REVIEW

File: tests/tests.rs [hunk 1]
```diff
@@ -26,17 +26,6 @@ fn test_simple() {
 }
 
 #[test]
-fn test_u16_index() {
```
Comment (bluss) [inline tests/tests.rs]: this was removed by mistake?

File: tests/tests.rs [hunk 2]
```diff
@@ -296,4 +284,9 @@ fn test_string() {
         Ok(())
     }();
     assert!(t.is_err());
+	
+	// Test `from` constructor
+    let u = ArrayString::<[_; 11]>::from(text).unwrap();
+	assert_eq!(&u, text);
```
Comment (bluss) [inline tests/tests.rs]: indentation

File: tests/tests.rs (hunk 1 above)
Comment (alexreg) [inline tests/tests.rs]: Yes, accident.

File: tests/tests.rs (hunk 2 above)
Comment (alexreg) [inline tests/tests.rs]: That’s my text editor (and personal preference) I’m afraid, sorry.

Comment (bluss) [inline tests/tests.rs]: 4 spaces on one line and 8 on the next? it must be a mistake

File: tests/tests.rs [hunk 3]
```diff
@@ -296,4 +295,9 @@ fn test_string() {
         Ok(())
     }();
     assert!(t.is_err());
+
+	// Test `from` constructor
+    let u = ArrayString::<[_; 11]>::from(text).unwrap();
+	assert_eq!(&u, text);
```
Comment (bluss) [inline tests/tests.rs]: can you fix the indentation on this line? Seems to be a spurious tab.

Comment (bluss) [inline tests/tests.rs]: Should be 4 spaces.

**Comment by bluss** [comment]
Thanks! I'll fixup the rest.

**Pull request merged**